### PR TITLE
Refine time slot bar and introducing minified variant

### PR DESCRIPTION
**Pull request by weiminlee95**
**Changes**

- Introduce "minified" variant
- Adjust default variant cell dimensions

- delete branch

**Changelog entry**

- Refine time-slot-bar and introducing minified variant

**Additional information**

- You may refer to https://www.figma.com/file/us90jOpWBahsqK2VAluPvD/Flagship-Design-System?type=design&node-id=7863%3A49200&mode=design&t=VYO3ARRrB99pOoyz-1

### CODE REVIEW

File: src/time-slot-bar/time-slot-bar.tsx [hunk 1]
```diff
@@ -116,15 +119,31 @@ export const TimeSlotBar = ({
...
+        const isStartHourEven = startTimeFormatted.hour() % 2 === 0;
+
         for (
             let currentTime = startTimeFormatted;
             currentTime.isBefore(endTimeFormatted);
             currentTime = currentTime.add(CELL_DURATION, "minute")
         ) {
             const isHour = currentTime.minute() === 0;
+
+            const isAlternateHour =
```
Comment (keithtxw) [inline src/time-slot-bar/time-slot-bar.tsx:131]: i find this logic abit hard to follow. Instead of using a const, could we extract it into a function instead? So `$isLongMarker` prop to use the function to get a boolean value

**Review by keithtxw — CHANGES_REQUESTED** [review]

File: src/time-slot-bar/time-slot-bar.tsx (hunk 1 above)
Comment (weiminlee95) [reply src/time-slot-bar/time-slot-bar.tsx]: Extracted to a separate helper function

File: stories/time-slot-bar/props-table.tsx [hunk 2]
```diff
@@ -15,6 +15,12 @@ const DATA: ApiTableSectionProps[] = [
                 description: "The class selector of the component",
                 propTypes: ["string"],
             },
+            {
+                name: "variant",
+                description: <>The variant type</>,
```
Comment (keithtxw) [inline stories/time-slot-bar/props-table.tsx:20]:
```suggestion
                description: <>The display variant of the <code>TimeSlotBar</code></>,
```

File: src/time-slot-bar/types.ts [hunk 3]
```diff
@@ -16,6 +18,8 @@ interface TimeSlotBarStyleAttributes {
 export interface TimeSlotBarProps {
     "data-testid"?: string;
     className?: string;
+    /** The time slot bar variant. Values: "minified" | "default" */
```
Comment (keithtxw) [inline src/time-slot-bar/types.ts:21]:
```suggestion
    /** The display variant of the component. Values: "minified" | "default" */
```

File: src/time-slot-bar/time-slot-bar.tsx [hunk 4]
```diff
@@ -148,26 +186,25 @@ export const TimeSlotBar = ({
         const slotWidth = TimeSlotBarHelper.calculateWidth(
             startTime,
             endTime,
-            CELL_WIDTH
+            cellWidth
         );
+
+        const isClickable = !!onClick && variant === "default";
```
Comment (keithtxw) [inline src/time-slot-bar/time-slot-bar.tsx:192]: So it is clickable only if you passed in onClick and it is default variant? Or actually if it is default variant it is already clickable?

**Review by keithtxw — CHANGES_REQUESTED** [review]

File: src/time-slot-bar/time-slot-bar.styles.tsx [hunk 5]
```diff
@@ -83,30 +112,46 @@ export const ArrowIconLeft = styled(ChevronLeftIcon)`
...
     display: inline-block;
-    width: ${CELL_WIDTH}px;
-    height: ${({ isHour }) => (isHour ? "1.25rem" : "0.625rem")};
+    width: ${({ $variant }) => `${getCellWidth($variant)}px`};
     position: relative;
-    border-left: 1px solid ${Color.Neutral[4]};
+    border-left: 1px solid ${Color.Neutral[2]};
+    ${(props) => {
+        let markerHeight = 0;
+        if (props.$variant === "default") {
```
Comment (keithtxw) [inline src/time-slot-bar/time-slot-bar.styles.tsx:143]: sorry can we do switch case for this

File: stories/time-slot-bar/time-slot-bar.stories.mdx [hunk 6]
```diff
@@ -63,6 +71,65 @@ import { TimeSlotBar } from "@lifesg/react-design-system/time-slot-bar";
     </Story>
 </Canvas>
 
+<Secondary>Minified Variant</Secondary>
+
+Here is an example of a minified variant of the component. This is useful for displaying a large number of time slots in a smaller view port.
```
Comment (keithtxw) [inline stories/time-slot-bar/time-slot-bar.stories.mdx:76]:
```suggestion
In cases where there is a requirement to display a large number of time slots in a smaller viewport, you can make use of the minified variant of the component. To do so, set `variant="minified"`.
```

**Review by keithtxw — CHANGES_REQUESTED** [review]
Sorry found some minor things to be fixed

File: src/time-slot-bar/time-slot-bar.tsx [hunk 7]
```diff
@@ -98,14 +101,38 @@ export const TimeSlotBar = ({
...
     };
 
+    /**
+     * Function to determine if a long time marker should be displayed
+     * For minified variant the long time marker is displayed every alternate hour
+     */
+    const shouldDisplayLongTimeMarker = (
+        currentTime,
+        isStartHourEven, // To determine if the first hour is even or odd for minified
+        variant
```
Comment (weili-govtech) [inline src/time-slot-bar/time-slot-bar.tsx:119]: can we declare the typing for each argument?

File: src/time-slot-bar/time-slot-bar.tsx [hunk 8]
```diff
@@ -195,17 +232,20 @@ export const TimeSlotBar = ({
...
                 startTime,
                 slotStartTime,
-                CELL_WIDTH
+                cellWidth
             );
 
+            const isClickable = clickable && variant === "default";
+
             return (
                 <>
```
Comment (weili-govtech) [inline src/time-slot-bar/time-slot-bar.tsx:246]: actually this fragment is the component that needs the `key` prop instead of the `Timeslot` component

File: src/time-slot-bar/time-slot-bar.tsx [hunk 9]
```diff
@@ -148,26 +186,25 @@ export const TimeSlotBar = ({
...
                     $bgColor2={backgroundColor2}
-                    $clickable={!!onClick}
-                    onClick={onClick}
-                />
-                <Border
-                    style={{
-                        left: `${slotWidth}px`,
-                    }}
+                    $clickable={isClickable}
+                    onClick={isClickable && onClick}
```
Comment (weili-govtech) [inline src/time-slot-bar/time-slot-bar.tsx:207]:
```suggestion
                    onClick={isClickable ? onClick : undefined}
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/time-slot-bar/time-slot-bar.tsx (hunk 8 above)
Comment (weili-govtech) [reply src/time-slot-bar/time-slot-bar.tsx]: just leaving a note because it's originally a fragment, it's possible to assign key by specifying the Fragment component like `<Fragment key={id}>`. but since using div does not break anything, I am approving the change.

**Review by weili-govtech — APPROVED** [review]

**Review by keithtxw — APPROVED** [review]
